### PR TITLE
Various fixes and test tweaks

### PR DESCRIPTION
**Pull request by cgillum**
Added a new test to validate #643, plus made a few other small tweaks that I discovered while updating docs.

Note the `DurableTaskOptions.cs` has a bunch of newline changes, which will make the diff harder to look at. You'll want to turn off whitespace comparisons for that.

### CODE REVIEW

File: test/Common/DurableTaskEndToEndTests.cs [hunk 1]
```diff
@@ -1115,6 +1115,36 @@ public async Task HandledActivityException(bool extendedSessions)
...
+            using (JobHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.HandledActivityException),
+                enableExtendedSessions: true))
+            {
+                await host.StartAsync();
+
+                string message = $"Failure ID: {Guid.NewGuid()}";
+
+                // Empty string input should result in ArgumentNullException in the orchestration code.
```
Comment (ConnorMcMahon) [inline test/Common/DurableTaskEndToEndTests.cs]: Is this comment meant for this test?

File: test/Common/TestActivities.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public static long GetFileSize([ActivityTrigger] DurableActivityContext ctx)
         public static void ThrowActivity([ActivityTrigger] DurableActivityContext ctx)
         {
             string message = ctx.GetInput<string>();
-            throw new Exception(message);
+            throw new InvalidOperationException(message);
```
Comment (ConnorMcMahon) [inline test/Common/TestActivities.cs]: Was this change required? I thought `SubOrchestrationThrow` calls `ThrowOrchestrator`, not `ThrowActivity`.

Comment (cgillum) [reply test/Common/TestActivities.cs]: This is not required. But I added it because at one point we were having issues where exceptions were not being deserialized to the correct type. Throwing a non-generic Exception would have helped us catch that, so I figure it's worth adding it now.

File: test/Common/DurableTaskEndToEndTests.cs (hunk 1 above)
Comment (cgillum) [reply test/Common/DurableTaskEndToEndTests.cs]: Nope. Copy/paste error. Good catch.
